src/price/price.service.ts

```diff
@@ -1,29 +1,61 @@
-import { Injectable } from '@nestjs/common'
-import { PrismaService } from 'src/prisma/prisma.service'
+import { Injectable, NotFoundException } from '@nestjs/common'
+import { PrismaService } from '../prisma/prisma.service'
 import { priceDto } from './dto'
 
 @Injectable()
 export class PriceService {
   constructor(private prisma: PrismaService) {}
-  async findOrCreate(data: priceDto) {
-    return this.prisma.price.upsert({
+
+  async findOrCreate(priceData: priceDto) {
+    const { bookId, platformId, date, productUrl } = priceData
+
+    let price = await this.prisma.price.findFirst({
       where: {
-        bookId_platformId: {
-          bookId: data.bookId,
-          platformId: data.platformId,
+        bookId,
+        platformId,
+        date,
+        productUrl,
+      },
+    })
+
+    if (!price) {
+      price = await this.prisma.price.create({
+        data: {
+          bookId,
+          platformId,
+          price: priceData.price,
+          date,
+          productUrl,
         },
+      })
+    }
```

Comment:
What is the reason for changing the findOrCreate method?  instead of using the upsert method?